tests/test_metrics.py

```diff
@@ -501,11 +530,16 @@ def test_collinearity_fixer(self):
 
             # Check if the correct metrics were trimmed
             for metric in scenario_data["metrics"]:
-                if str(metric) in scenario_data["expected_trimmed_metrics"]:
-                    self.assertEqual(collection.get_metric(str(metric)).base_idx, 1)
+                if str(metric) in scenario_data["expected_trimmed_metrics"].keys():
+                    self.assertEqual(set(collection.get_metric(str(metric))._indices_to_ignore),
+                                     set(scenario_data["expected_trimmed_metrics"][str(metric)]))
                 elif hasattr(metric,
-                             "base_idx"):  # Only metrics with a base_idx can be trimmed. The others aren't tested
-                    self.assertEqual(collection.get_metric(str(metric)).base_idx, 0)
+                             "_indices_to_ignore"):  # Only metrics with a base_idx can be trimmed. The others aren't tested
+                    self.assertEqual(collection.get_metric(str(metric))._indices_to_ignore, [])
+                elif str(metric) in scenario_data["expected_eliminated_metrics"]:
+                    self.assertTrue(str(metric) not in [str(m) for m in collection.metrics])
```

Comment:
If testing for metrics = [NumEdges(), NumEdges()], this test will always fail because the eliminated metric is called NumEdges(), and so is the remaining metric.